### PR TITLE
Clickable hyperlinks in Spyglass Build log view

### LINKED ISSUE #23311: Clickable hyperlinks in Spyglass Build log view
**What would you like to be added**:
Clickable hyperlinks for Spyglass Build log view.

**Why is this needed**:
Whenever a developer looks at the build logs, sometimes there are links that can be used for further debugging.
It would be faster if the user could click on the link and it would open it instead of copying it to the browser URL input.

for example for the following links:
```
INFO[2021-08-19T10:48:21Z] Loading configuration from https://config.ci.openshift.org for openshift/assisted-service@master 
INFO[2021-08-19T10:48:21Z] Resolved source https://github.com/openshift/assisted-service to master@fd2dcd0a, merging: #2423 856171be @YuviGold 
INFO[2021-08-19T10:48:21Z] Resolved release candidate to quay.io/openshift-release-dev/ocp-release@sha256:e64c04c41ae7717fff4b341987ac37c313045d4c3aa7bb8c6bfe8bf8540a5025 
INFO[2021-08-19T10:48:22Z] Using namespace https://console.build01.ci.openshift.org/k8s/cluster/projects/ci-op-iff5f2pm 
```

### PR DESCRIPTION
**Pull request by YuviGold**
Resolves #23311

Solution taken from https://stackoverflow.com/questions/29055828/regex-to-make-links-clickable-in-only-a-href-and-not-img-src
Probably not the best solution, but it works for now until we will find a better one. :)

Screenshot from 2021-09-01 02-09-50

### CODE REVIEW

File: prow/cmd/deck/static/spyglass/lens.ts [hunk 1]
```diff
@@ -189,6 +191,21 @@ class SpyglassImpl implements Spyglass {
...
 
+ private setLink(match: string, attr: string): string {
+    if (typeof attr != 'undefined') {
+        return match;
+    }
+    return '</span><a target="_blank" href="' + match + '">' + match +'</a><span>';
+  }
+
+  private createHyperlinks(parent: Element): void {
+    var re = /((?:href|src)=")?(\b(https?|ftp|file):\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|])/ig;
```
Comment (chaodaiG) [inline prow/cmd/deck/static/spyglass/lens.ts:202]: this is a 😮  regex, can you link the stack overflow post here please?

Comment (YuviGold) [reply prow/cmd/deck/static/spyglass/lens.ts]: Done

**Review by chaodaiG — APPROVED** [review]

**Comment by howardjohn** [comment]
This *might* have broken https://github.com/istio/test-infra/issues/3538. No clue if it actually is the root cause, just a guess best on similar code path

**Pull request merged**